### PR TITLE
feat: implement download project via curl (#3227)

### LINKED ISSUE #3227: [HCA DB] Implement download project via curl
#### Depends On
- [x] https://github.com/DataBiosphere/findable-ui/issues/12

#### Approach
* Configure findable-ui functionality.

### PR DESCRIPTION
**Pull request by frano-m**
### Ticket
Closes #3227.

### Reviewers
@NoopDog.

### Changes
- Added download project via curl.
- Refactored file manifest hooks for reuse with export to terra.
- Updated `@clevercanary/data-explorer-ui` to version `v0.25.0`.

### CODE REVIEW

File: explorer/app/components/Detail/components/Export/common/entities.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { SelectCategory } from "@clevercanary/data-explorer-ui/lib/common/entities";
+
+/**
+ * Set of possible values for the export filter key.
+ */
+export enum ExportFilterKey {
```
Comment (frano-m) [inline explorer/app/components/Detail/components/Export/common/entities.ts:6]: Refactored for re-use.

File: explorer/app/components/Detail/components/Export/components/ExportEntityToTerraForm/exportEntityToTerraForm.tsx [hunk 2]
```diff
@@ -1,24 +1,7 @@
...
-
-export interface ExportSelectCategory extends Omit<SelectCategory, "values"> {
-  values: string[];
-}
-
-export type ExportFilterKeyExportCategory = Map<
-  ExportFilterKey,
-  ExportSelectCategory
->;
+import { ExportFilterKeyExportCategory } from "../../common/entities";
```
Comment (frano-m) [inline explorer/app/components/Detail/components/Export/components/ExportEntityToTerraForm/exportEntityToTerraForm.tsx:4]: Refactored for re-use.

File: explorer/app/hooks/azul/useFileManifestRequestParams.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ export const useExportEntityToTerraRequestParams = (): URLSearchParams => {
   // Build request params.
   return new URLSearchParams({
     [AZUL_PARAM.CATALOG]: catalog,
-    [EXPORT_TO_TERRA_PARAM.FORMAT]: EXPORT_TO_TERRA_FORMAT.PFB,
+    format: manifestFormat,
```
Comment (frano-m) [inline explorer/app/hooks/azul/useFileManifestRequestParams.ts:24]: Refactored for re-use. Manifest format now passed in as a prop.

File: explorer/app/viewModelBuilders/azul/anvil-cmg/common/viewModelBuilders.ts [hunk 4]
```diff
@@ -351,8 +351,8 @@ export const buildExportEntityToTerra = (
       }),
     ExportToTerra: MDX.ExportToTerra,
     ExportToTerraSuccess: MDX.ExportToTerraSuccess,
-    useExportParams: useExportEntityToTerraRequestParams,
-    useExportRequestURL: useExportEntityToTerraRequestURL,
+    useExportParams: useFileManifestRequestParams,
```
Comment (frano-m) [inline explorer/app/viewModelBuilders/azul/anvil-cmg/common/viewModelBuilders.ts:354]: Refactored.

File: explorer/package.json [hunk 5]
```diff
@@ -32,7 +32,7 @@
     "test:anvil-catalog": "playwright test -c playwright_anvil-catalog.config.ts"
   },
   "dependencies": {
-    "@clevercanary/data-explorer-ui": "0.22.0",
+    "@clevercanary/data-explorer-ui": "0.25.0",
```
Comment (frano-m) [inline explorer/package.json:35]: Package upgrade!

**Review by NoopDog — APPROVED** [review]

**Pull request merged**